operator/v1/types_csi_cluster_driver.go

```diff
@@ -369,6 +369,24 @@ type VSphereCSIDriverConfigSpec struct {
 	// +openshift:enable:FeatureGate=VSphereDriverConfiguration
 	// +optional
 	GranularMaxSnapshotsPerBlockVolumeInVVOL *uint32 `json:"granularMaxSnapshotsPerBlockVolumeInVVOL,omitempty"`
+
+	// maxAllowedBlockVolumesPerNode is a configuration parameter that allows setting custom value for limit of
+	// PersistentVolumes attached to a node. In vSphere version 7 this limit was set to 59 by default, however
+	// in vSphere version 8 this limit was increased to 255. For more details see: https://configmax.broadcom.com/guest?vmwareproduct=vSphere&release=vSphere%208.0&categories=3-0
+	// Before increasing this value above 59 the cluster administrator needs to ensure that every node forming the
+	// cluster is updated to ESXi version 8 or higher and that all nodes are running the same version.
+	// According to CSI protocol the cluster orchestrator can decide on default value if the limit is not set or
+	// set to 0: https://github.com/container-storage-interface/spec/blob/98819c45a37a67e0cd466bd02b813faf91af4e45/csi.proto#L1679-L1684
+	// Since cluster orchestrator could change its defaults this field is set to a default value that matches the
+	// original limit for vSphere version 7.
+	// Setting this value to 0 would effectively prevent CSI controller from publishing any CSI volumes to the node.
```

Comment:
Does not setting the value and setting it to 0 do the same thing?

Reply:
For node installer manager and scheduler code `0` and nil is treated the same - so yes. I've misunderstood this previously - see if the updated version makes more sense.

This also means we should probably not have this as optional pointer field because that could unexpectedly remove the limit if the field is unset. So we should set a default that's not zero/nil in all cases to prevent this and adding a default here makes more sense than letting operator detect the unset field and setting some default there.